iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs

```diff
@@ -85,7 +66,11 @@ private async Task RefreshLoopAsync(DateTime expiry, CancellationToken cancellat
             {
                 if (Logging.IsEnabled) Logging.Info(this, expiry, $"Before {nameof(RefreshLoopAsync)}");
 
-                if (waitTime.Milliseconds > 0)
+                if (waitTime < MinWaitTime)
```

Comment:
There shouldn't be a min wait time. Having one may unnecessarily delay a refresh which could take a very long time within the process of getting/sending the token.

Reply:
I added one second min interval delay to avoid infinite loop.

Reply:
In that case, we should include the refresh time (start a Stopwatch and check to see how long that took). This code will _add_ a second even if the refresh took minutes.